javascript/javascript1/week3/CactusIO-Interactive.js

```diff
@@ -0,0 +1,107 @@
+/* Create a variable called activities that stores all activities of the day.  
+Create a function called addActivity. 
+It should have three parameters: date, activity and duration. 
+The date should be a string, the activity a string and the duration a number. To save the activity push an object that contains the date, the activity and the duration to the activities variable.
+
+Create a function called showStatus. This function should use the activities variable and return a string saying the following: "You have added 3 activities. They amount to 78 min. of usage" 
+Now what happens if we call showStatus and activities is empty? We need to take that into consideration: If activities is empty log out a string that says: "Add some activities before calling showStatus"
+
+When showStatus is called and the users usage is above the limit he set. Log out the following string: "You have reached your limit, no more smartphoning for you!"
+
+Lets improve the addActivity, so that we dont need to specify the date, but the function automatically figures out what the date is. Check out this link: https://stackoverflow.com/a/34015511
+
+Improve the showStatus function by only showing the number of actitivies for that specific day.
+Create a function for calculating the activity a user has spent the most time on.*/
+
+
+const activities = [];
+
+function addActivity(date, activity, duration) {
+    date = new Date(date);
+
+    if (!date || !activity || !duration) {
+        console.log("Please provide a date, activity, and duration");
+        return;
+    }
+
+
+    let formattedDate;
+    if (typeof date === "string") {
+        formattedDate = date;
```

Comment:
Is this condition necessary? Because early you're casting the string to a `Date` object.

Reply:
Or maybe you forgot to remove line 20